# Introduction
The purpose of this notebook is to create a file of sites to which we can add GPS coordinates. We will first check the correlations between the sites previously curated, and those emailed by Olivo today

In [1]:
%run _standard_imports.ipynb

python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
numpy 1.11.2
scipy 0.18.1
pandas 0.19.0
numexpr 2.6.1
pysam 0.8.4
petl 1.1.0
petlx 1.0.3
vcf 0.6.8
vcfnp 2.2.0
h5py 2.6.0
tables 3.3.0


In [27]:
panoptes_final_metadata_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_panoptes_samples_final_20170124.txt.gz"
olivo_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/samplesMeta6-toRP.xlsx"
comparison_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_olivo_comparison_20170206.xlsx"
site_comparison_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_olivo_site_comparison_20170206.xlsx"
year_comparison_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_olivo_year_comparison_20170206.xlsx"
sites_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_sites_20170204.xlsx"
hdf_fn = "/nfs/team112_internal/production/release_build/Pf/6_0_release_packages/hdf5/Pf_60.h5"

In [3]:
tbl_panoptes_samples_final = (
    etl
    .fromtsv(panoptes_final_metadata_fn)
    .convertnumbers()
)
print(len(tbl_panoptes_samples_final.data()))
print(len(tbl_panoptes_samples_final.distinct('Sample').data()))
tbl_panoptes_samples_final

7182
7182


+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| Sample     | OxfordSrcCode | OxfordDonorCode | Individual_ID | AlfrescoStudyCode   | Type  | InV5.1 | InV5   | SubContinentCode | SubContinentName | RegionCode | CountryCode | CountryName | Site      | Year | Date | HasDuplicate | DiscardAsDuplicate | run_accessions                            | pc_pass_missing     | pc_genome_covered_at_1x | Year_source | sort_AlfrescoStudyCode | IndividualGroup |
+============+===============+=================+===============+=====================+=======+========+========+==================+==================+============+=============+=============+===========+======+======+==============+====================+===========================================+=====================+=========================+=============+========================+=================+
| 'PM0001-C' | '215C'        | '215C'          | '215C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012350,ERR012360,ERR012305'           | 0.15139786050931692 |                  0.9696 | 'study'     | '1001-PF-ML-DJIMDE'    |               1 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0002-C' | '217C'        | '217C'          | '217C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012520,ERR012331,ERR012327'           | 0.28678549872221293 |                   0.864 | 'study'     | '1001-PF-ML-DJIMDE'    |               2 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0003-C' | '520C'        | '520C'          | '520C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012492,ERR012457,ERR012532'           |  0.1662372060768727 |                  0.9586 | 'study'     | '1001-PF-ML-DJIMDE'    |               3 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0004-C' | 'A336TPI7C'   | 'A336TPI7C'     | 'A336TPI7C'   | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR015438,ERR015329,ERR015422,ERR015315' | 0.11811536041300132 |                  0.9648 | 'study'     | '1001-PF-ML-D

In [4]:
tbl_olivo = (
    etl
    .fromxlsx(olivo_fn)
)
print(len(tbl_olivo.data()))
print(len(tbl_olivo.distinct('Sample').data()))
tbl_panoptes_samples_final

7414
7414


+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| Sample     | OxfordSrcCode | OxfordDonorCode | Individual_ID | AlfrescoStudyCode   | Type  | InV5.1 | InV5   | SubContinentCode | SubContinentName | RegionCode | CountryCode | CountryName | Site      | Year | Date | HasDuplicate | DiscardAsDuplicate | run_accessions                            | pc_pass_missing     | pc_genome_covered_at_1x | Year_source | sort_AlfrescoStudyCode | IndividualGroup |
+============+===============+=================+===============+=====================+=======+========+========+==================+==================+============+=============+=============+===========+======+======+==============+====================+===========================================+=====================+=========================+=============+========================+=================+
| 'PM0001-C' | '215C'        | '215C'          | '215C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012350,ERR012360,ERR012305'           | 0.15139786050931692 |                  0.9696 | 'study'     | '1001-PF-ML-DJIMDE'    |               1 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0002-C' | '217C'        | '217C'          | '217C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012520,ERR012331,ERR012327'           | 0.28678549872221293 |                   0.864 | 'study'     | '1001-PF-ML-DJIMDE'    |               2 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0003-C' | '520C'        | '520C'          | '520C'        | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR012492,ERR012457,ERR012532'           |  0.1662372060768727 |                  0.9586 | 'study'     | '1001-PF-ML-DJIMDE'    |               3 |
+------------+---------------+-----------------+---------------+---------------------+-------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+-------------------------------------------+---------------------+-------------------------+-------------+------------------------+-----------------+
| 'PM0004-C' | 'A336TPI7C'   | 'A336TPI7C'     | 'A336TPI7C'   | '1001-PF-ML-DJIMDE' | 'DNA' | 'True' | 'True' | 'WAF'            | 'West Africa'    | 'ML'       | 'ML'        | 'Mali'      | 'Kolle'   | 2007 | 2007 | 'False'      | 'False'            | 'ERR015438,ERR015329,ERR015422,ERR015315' | 0.11811536041300132 |                  0.9648 | 'study'     | '1001-PF-ML-D

In [28]:
tbl_comparison = (
    tbl_panoptes_samples_final
    .leftjoin(tbl_olivo, key='Sample', rprefix='Olivo_')
    .cut(['Sample', 'AlfrescoStudyCode', 'Site', 'Year', 'Olivo_Location', 'Olivo_Year'])
)
tbl_comparison.toxlsx(comparison_fn)


In [29]:
tbl_comparison = etl.fromxlsx(comparison_fn)


In [30]:
(
    tbl_comparison
    .valuecounts('Olivo_Location', 'Site')
    .cutout('frequency')
    .displayall()
)

Olivo_Location 
 Site 
 count 
 
 
 
 
 Pursat 
 Pursat 
 540 
 
 
 Kassena 
 Navrongo 
 522 
 
 
 Wang Pha 
 Mae Sot 
 449 
 
 
 Kinshasa 
 Kinshasa 
 366 
 
 
 Chikwawa 
 Chikwawa 
 300 
 
 
 Ratanakiri 
 Ratanakiri 
 243 
 
 
 Mawker Thai 
 Mae Sot 
 222 
 
 
 Preah Vihear 
 Preah Vihear 
 210 
 
 
 Mae La 
 Mae Sot 
 176 
 
 
 Bamako 
 Bamako 
 164 
 
 
 Navrongo 
 Navrongo 
 164 
 
 
 Tanga 
 Tanga 
 160 
 
 
 Pailin 
 Pailin 
 157 
 
 
 - 
 Navrongo 
 155 
 
 
 Buea 
 Buea 
 145 
 
 
 Phuoc Long 
 Phuoc Long 
 138 
 
 
 Faladie 
 Faladje 
 137 
 
 
 Nzerekore 
 Nzerekore 
 137 
 
 
 Binh Phuoc 
 Binh Phuoc 
 126 
 
 
 Basse 
 Basse 
 124 
 
 
 Brikama 
 Brikama 
 123 
 
 
 Homel 
 Homel 
 102 
 
 
 Accra 
 Accra 
 101 
 
 
 Bago 
 Bago 
 94 
 
 
 Attapeu 
 Attapeu 
 86 
 
 
 Nachingwea 
 Nachingwea 
 79 
 
 
 Kilifi 
 Kilifi 
 71 
 
 
 Lab 
 Lab 
 70 
 
 
 - 
 Timika 
 70 
 
 
 Tasanh 
 Tasanh 
 65 
 
 
 Muleba 
 Muleba 
 61 
 
 
 Kintampo 
 Kintampo 
 61 
 
 
 Faranah 
 Faranah 
 60 
 
 
 Pikine 
 Pikine 
 60 
 
 
 Bobo-Dioulasso 
 Bobo-Dioulasso 
 56 
 
 
 Madang 
 Madang 
 55 
 
 
 Thabeikkyin 
 Thabeikkyin 
 54 
 
 
 East Sepik 
 East Sepik 
 53 
 
 
 Bamako 
 Nioro du Sahel 
 52 
 
 
 Kolle 
 Kolle 
 51 
 
 
 Ramu 
 Ramu 
 51 
 
 
 Zomba 
 Zomba 
 51 
 
 
 Kawthaung 
 Kawthaung 
 51 
 
 
 Xepon 
 Xepon 
 45 
 
 
 Mae Khon Kaen 
 Mae Sot 
 45 
 
 
 Mae Sot 
 Mae Sot 
 44 
 
 
 Bandarban 
 Bandarban 
 42 
 
 
 Morogoro 
 Morogoro 
 34 
 
 
 Bagdary 
 Bagdary 
 34 
 
 
 Kisumu 
 Kisumu 
 34 
 
 
 - 
 Nema 
 33 
 
 
 Abobo 
 Abobo 
 31 
 
 
 Milne Bay 
 Milne Bay 
 30 
 
 
 Myitkyina 
 Myitkyina 
 28 
 
 
 Si Sa Ket 
 Sisakhet 
 28 
 
 
 Ranong 
 Ranong 
 27 
 
 
 - 
 Pikine 
 26 
 
 
 Kombewa 
 Kombewa 
 26 
 
 
 Pyin Oo Lwin 
 Pyin Oo Lwin 
 23 
 
 
 - 
 Kobeni 
 23 
 
 
 Mile 16 
 Buea 
 23 
 
 
 Papua 
 Timika 
 22 
 
 
 Faladje 
 Faladje 
 22 
 
 
 - 
 Selibaby 
 21 
 
 
 Yopougon 
 Yopougon 
 20 
 
 
 Koumassi 
 Koumassi 
 19 
 
 
 - 
 West Arsi Zone 
 18 
 
 
 Maevatanana 
 Maevatanana 
 18 
 
 
 Muheza 
 Muheza 
 16 
 
 
 - 
 Njaiyel 
 16 
 
 
 - 
 Amhara region 
 15 
 
 
 Falaje 
 Faladje 
 14 
 
 
 - 
 Madina Samako 
 14 
 
 
 Mutengene 
 Buea 
 14 
 
 
 Apac 
 Apac 
 14 
 
 
 - 
 Loreto 
 13 
 
 
 Molyko 
 Buea 
 13 
 
 
 Muea 
 Buea 
 11 
 
 
 Iquitos 
 Iquitos, Loreto Province 
 11 
 
 
 Bandiagara 
 Bandiagara 
 9 
 
 
 - 
 Aioun 
 9 
 
 
 Ilorin 
 Ilorin 
 8 
 
 
 Tumaco 
 Tumaco 
 7 
 
 
 Toamasina 
 Antsohihy 
 6 
 
 
 Tole 
 Buea 
 5 
 
 
 Mile 14 
 Buea 
 5 
 
 
 Iquitos 
 Lab 
 5 
 
 
 Mile 15 
 Buea 
 5 
 
 
 Guapi 
 Guapi 
 4 
 
 
 - 
 Ghana returning traveller 
 3 
 
 
 Bunduma 
 Buea 
 3 
 
 
 Quibdo 
 Quibdo 
 3 
 
 
 Buenaventura 
 Buenaventura 
 3 
 
 
 Ombe 
 Buea 
 2 
 
 
 - 
 Mali returning traveller 
 2 
 
 
 Checkpoint 
 Buea 
 2 
 
 
 Bomaka 
 Buea 
 2 
 
 
 - 
 Kenya returning traveller 
 2 
 
 
 - 
 Uganda returning traveller 
 2 
 
 
 Nyasoso 
 Buea 
 1 
 
 
 Great Soppo 
 Buea 
 1 
 
 
 Bolifamba 
 Buea 
 1 
 
 
 Yagaum, Madang 
 Madang 
 1 
 
 
 Mile 18 
 Buea 
 1 
 
 
 Toamasina 
 Farafangana 
 1 
 
 
 Ouezzin-ville; Bobo-Dioulasso 
 Bobo-Dioulasso 
 1 
 
 
 Buea Town 
 Buea 
 1 
 
 
 - 
 Lab 
 1 
 
 
 Likomba 
 Buea 
 1 
 
 
 - 
 Kombewa 
 1 
 
 
 - 
 Mozambique returning traveller 
 1 
 
 
 Bojongo 
 Buea 
 1 
 
 
 Bokova 
 Buea 
 1 
 
 
 Soppo 
 Buea 
 1 
 
 
 West_Arsi 
 West Arsi Zone 
 1

In [31]:
(
    tbl_comparison
    .valuecounts('Olivo_Location', 'Site')
    .cutout('frequency')
    .toxlsx(site_comparison_fn)
)

In [32]:
(
    tbl_comparison
    .valuecounts('Olivo_Year', 'Year')
    .cutout('frequency')
    .displayall()
)

Olivo_Year 
 Year 
 count 
 
 
 
 
 - 
 2013 
 930 
 
 
 2013 
 2013 
 831 
 
 
 - 
 2011 
 799 
 
 
 - 
 2014 
 743 
 
 
 2011 
 2011 
 551 
 
 
 2012 
 2012 
 511 
 
 
 2008 
 2008 
 470 
 
 
 - 
 2010 
 390 
 
 
 - 
 2012 
 303 
 
 
 2010 
 2010 
 212 
 
 
 - 
 2015 
 205 
 
 
 - 
 2009 
 188 
 
 
 - 
 2008 
 174 
 
 
 - 
 2007 
 169 
 
 
 2014 
 2014 
 119 
 
 
 2003 
 2003 
 86 
 
 
 2004 
 2004 
 76 
 
 
 2007 
 2007 
 71 
 
 
 Lab 
 Lab 
 70 
 
 
 2002 
 2002 
 52 
 
 
 2005 
 2005 
 51 
 
 
 2013 
 2012 
 48 
 
 
 2009 
 2009 
 40 
 
 
 2001 
 2001 
 37 
 
 
 2012 
 2013 
 18 
 
 
 - 
 2011† 
 12 
 
 
 2013 
 2014 
 11 
 
 
 - 
 Lab 
 6 
 
 
 2011 
 2010 
 3 
 
 
 2006 
 2006 
 2 
 
 
 - 
 2004 
 1 
 
 
 - 
 2003 
 1 
 
 
 - 
 2001 
 1 
 
 
 2012 
 2011 
 1

In [33]:
(
    tbl_comparison
    .valuecounts('Olivo_Year', 'Year')
    .cutout('frequency')
    .toxlsx(year_comparison_fn)
)

In [18]:
(
    tbl_panoptes_samples_final
    .valuecounts('Site', 'CountryCode')
    .sort(('CountryCode', 'Site'))
    .cut(['Site', 'CountryCode'])
    .toxlsx(sites_fn)
)


# Next steps
After creating the above file, it was manually populated using http://latitude.to/ and other information such as SIS and Google maps. Final version (pf_6_0_sites_20170206.xlsx) sent to Ben.

In [20]:
(
    tbl_panoptes_samples_final
    .valuecounts('Site', 'CountryCode', 'CountryName')
    .sort(('CountryCode', 'Site'))
    .displayall()
)

Site 
 CountryCode 
 CountryName 
 count 
 frequency 
 
 
 
 
 Bandarban 
 BD 
 Bangladesh 
 42 
 0.005847953216374269 
 
 
 Ramu 
 BD 
 Bangladesh 
 51 
 0.007101086048454469 
 
 
 Bobo-Dioulasso 
 BF 
 Burkina Faso 
 57 
 0.007936507936507936 
 
 
 Homel 
 BJ 
 Benin 
 102 
 0.014202172096908938 
 
 
 Kinshasa 
 CD 
 Congo DR 
 366 
 0.050960735171261484 
 
 
 Abobo 
 CI 
 Ivory Coast 
 31 
 0.004316346421609579 
 
 
 Koumassi 
 CI 
 Ivory Coast 
 19 
 0.0026455026455026454 
 
 
 Yopougon 
 CI 
 Ivory Coast 
 20 
 0.00278473962684489 
 
 
 Buea 
 CM 
 Cameroon 
 239 
 0.033277638540796434 
 
 
 Buenaventura 
 CO 
 Colombia 
 3 
 0.0004177109440267335 
 
 
 Guapi 
 CO 
 Colombia 
 4 
 0.000556947925368978 
 
 
 Quibdo 
 CO 
 Colombia 
 3 
 0.0004177109440267335 
 
 
 Tumaco 
 CO 
 Colombia 
 7 
 0.0009746588693957114 
 
 
 Amhara region 
 ET 
 Ethiopia 
 15 
 0.0020885547201336674 
 
 
 West Arsi Zone 
 ET 
 Ethiopia 
 19 
 0.0026455026455026454 
 
 
 Accra 
 GH 
 Ghana 
 101 
 0.014062935115566694 
 
 
 Ghana returning traveller 
 GH 
 Ghana 
 3 
 0.0004177109440267335 
 
 
 Kintampo 
 GH 
 Ghana 
 61 
 0.008493455861876914 
 
 
 Navrongo 
 GH 
 Ghana 
 841 
 0.11709830130882762 
 
 
 Basse 
 GM 
 Gambia 
 124 
 0.017265385686438317 
 
 
 Brikama 
 GM 
 Gambia 
 123 
 0.017126148705096073 
 
 
 Madina Samako 
 GM 
 Gambia 
 14 
 0.001949317738791423 
 
 
 Njaiyel 
 GM 
 Gambia 
 16 
 0.002227791701475912 
 
 
 Faranah 
 GN 
 Guinea 
 60 
 0.00835421888053467 
 
 
 Nzerekore 
 GN 
 Guinea 
 137 
 0.019075466443887495 
 
 
 Timika 
 ID 
 Indonesia 
 92 
 0.012809802283486493 
 
 
 Kenya returning traveller 
 KE 
 Kenya 
 2 
 0.000278473962684489 
 
 
 Kilifi 
 KE 
 Kenya 
 71 
 0.009885825675299359 
 
 
 Kisumu 
 KE 
 Kenya 
 34 
 0.004734057365636313 
 
 
 Kombewa 
 KE 
 Kenya 
 27 
 0.0037593984962406013 
 
 
 Pailin 
 KH 
 Cambodia 
 157 
 0.021860206070732385 
 
 
 Preah Vihear 
 KH 
 Cambodia 
 210 
 0.029239766081871343 
 
 
 Pursat 
 KH 
 Cambodia 
 540 
 0.07518796992481203 
 
 
 Ratanakiri 
 KH 
 Cambodia 
 243 
 0.03383458646616541 
 
 
 Tasanh 
 KH 
 Cambodia 
 65 
 0.009050403787245892 
 
 
 Attapeu 
 LA 
 Laos 
 86 
 0.011974380395433026 
 
 
 Xepon 
 LA 
 Laos 
 45 
 0.006265664160401002 
 
 
 Lab 
 Lab 
 Lab 
 76 
 0.010582010582010581 
 
 
 Antsohihy 
 MG 
 Madagascar 
 6 
 0.000835421888053467 
 
 
 Farafangana 
 MG 
 Madagascar 
 1 
 0.0001392369813422445 
 
 
 Maevatanana 
 MG 
 Madagascar 
 18 
 0.002506265664160401 
 
 
 Bamako 
 ML 
 Mali 
 164 
 0.022834864940128097 
 
 
 Bandiagara 
 ML 
 Mali 
 9 
 0.0012531328320802004 
 
 
 Faladje 
 ML 
 Mali 
 173 
 0.024087997772208297 
 
 
 Kolle 
 ML 
 Mali 
 51 
 0.007101086048454469 
 
 
 Mali returning traveller 
 ML 
 Mali 
 2 
 0.000278473962684489 
 
 
 Nioro du Sahel 
 ML 
 Mali 
 52 
 0.007240323029796714 
 
 
 Bago 
 MM 
 Myanmar 
 94 
 0.013088276246170982 
 
 
 Kawthaung 
 MM 
 Myanmar 
 51 
 0.007101086048454469 
 
 
 Myitkyina 
 MM 
 Myanmar 
 28 
 0.003898635477582846 
 
 
 Pyin Oo Lwin 
 MM 
 Myanmar 
 23 
 0.0032024505708716233 
 
 
 Thabeikkyin 
 MM 
 Myanmar 
 54 
 0.007518796992481203 
 
 
 Aioun 
 MR 
 Mauritania 
 9 
 0.0012531328320802004 
 
 
 Kobeni 
 MR 
 Mauritania 
 23 
 0.0032024505708716233 
 
 
 Nema 
 MR 
 Mauritania 
 33 
 0.004594820384294068 
 
 
 Selibaby 
 MR 
 Mauritania 
 21 
 0.0029239766081871343 
 
 
 Chikwawa 
 MW 
 Malawi 
 300 
 0.04177109440267335 
 
 
 Zomba 
 MW 
 Malawi 
 51 
 0.007101086048454469 
 
 
 Mozambique returning traveller 
 MZ 
 Mozambique 
 1 
 0.0001392369813422445 
 
 
 Bagdary 
 NG 
 Nigeria 
 34 
 0.004734057365636313 
 
 
 Ilorin 
 NG 
 Nigeria 
 8 
 0.001113895850737956 
 
 
 Iquitos, Loreto Province 
 PE 
 Peru 
 11 
 0.0015316067947646894 
 
 
 Loreto 
 PE 
 Peru 
 13 
 0.0018100807574491784 
 
 
 East Sepik 
 PG 
 Papua New Guinea 
 53 
 0.007379560011138958 
 
 
 Madang 
 PG 
 Papua New Guinea 
 56 
 0.007797270955165692 
 
 
 Milne Bay 
 PG 
 Papua New Guinea 
 30 
 0.004177109440267335 
 
 

In [53]:
(
    tbl_panoptes_samples_final
    .selectin('IndividualGroup', problematic_groups)
    .cut(['Sample', 'OxfordSrcCode', 'Individual_ID', 'AlfrescoStudyCode', 'Site', 'Year', 'DiscardAsDuplicate', 'IndividualGroup'])
    .displayall()
)

Sample 
 OxfordSrcCode 
 Individual_ID 
 AlfrescoStudyCode 
 Site 
 Year 
 DiscardAsDuplicate 
 IndividualGroup 
 
 
 
 
 PH1025-C 
 163-KH1-032 
 163-KH1-032 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 False 
 1939 
 
 
 PH1090-C 
 163-KH1-032RME 
 163-KH1-032 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2013 
 True 
 1939 
 
 
 PH1347-C 
 163-KH1-032-CF11-adap 
 163-KH1-032 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1939 
 
 
 PH1381-C 
 163-KH1-032-adap 
 163-KH1-032 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1939 
 
 
 PH1026-C 
 163-KH1-033 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1940 
 
 
 PH1091-C 
 163-KH1-033RME 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2013 
 True 
 1940 
 
 
 PH1329-C 
 163-KH1-033-CF11-adap 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1940 
 
 
 PH1336-C 
 163-KH1-033RME-CF11-adap 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 False 
 1940 
 
 
 PH1362-C 
 163-KH1-033-adap 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1940 
 
 
 PH1369-C 
 163-KH1-033RME-adap 
 163-KH1-033 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2012 
 True 
 1940 
 
 
 PH1205-C 
 163-KH1-062 
 163-KH1-062 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2013 
 False 
 1962 
 
 
 PH1266-C 
 163-KH1-062RME 
 163-KH1-062 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2013 
 True 
 1962 
 
 
 PH1400-C 
 163-KH1-062-2-Filter 
 163-KH1-062 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2014 
 True 
 1962 
 
 
 PH1400-CW2 
 163-KH1-062-2 
 163-KH1-062 
 1044-PF-KH-FAIRHURST 
 Pursat 
 2015 
 True 
 1962 
 
 
 PH1039-C 
 163-KH2-012 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2060 
 
 
 PH1098-C 
 163-KH2-012RME 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2013 
 True 
 2060 
 
 
 PH1348-C 
 163-KH2-012RME-CF11-adap 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2060 
 
 
 PH1355-C 
 163-KH2-012-CF11-adap 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 False 
 2060 
 
 
 PH1382-C 
 163-KH2-012RME-adap 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2060 
 
 
 PH1389-C 
 163-KH2-012-adap 
 163-KH2-012 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2060 
 
 
 PH1099-C 
 163-KH2-021RME 
 163-KH2-021 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2013 
 True 
 2067 
 
 
 PH1349-C 
 163-KH2-021RME-CF11-adap 
 163-KH2-021 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2067 
 
 
 PH1353-C 
 163-KH2-021-CF11-adap 
 163-KH2-021 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2067 
 
 
 PH1383-C 
 163-KH2-021RME-adap 
 163-KH2-021 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 True 
 2067 
 
 
 PH1387-C 
 163-KH2-021-adap 
 163-KH2-021 
 1044-PF-KH-FAIRHURST 
 Preah Vihear 
 2012 
 False 
 2067 
 
 
 PH1316-C 
 163-KH3-062 
 163-KH3-062 
 1044-PF-KH-FAIRHURST 
 Ratanakiri 
 2013 
 False 
 2160 
 
 
 PH1322-C 
 163-KH3-062RME 
 163-KH3-062 
 1044-PF-KH-FAIRHURST 
 Ratanakiri 
 2014 
 True 
 2160 
 
 
 PJ0161-C 
 UNKNOWN_UFC_104 
 UFC_104 
 1155-PF-ID-PRICE 
 Timika 
 2011† 
 True 
 6614 
 
 
 PJ0183-C 
 UFC_104 
 UFC_104 
 1155-PF-ID-PRICE 
 Timika 
 2012 
 False 
 6614 
 
 
 PJ0134-C 
 UNKNOWN_UFM_32 
 UFM_32 
 1155-PF-ID-PRICE 
 Timika 
 2011† 
 False 
 6687 
 
 
 PJ0253-C 
 UFM_32 
 UFM_32 
 1155-PF-ID-PRICE 
 Timika 
 2011 
 True 
 6687

# Duplicate samples conclusion
There are a few samples from study 1044 that look like probable recrudescences. Two sample pairs from 1155 with different dates are due to assigned samples to 2011† in elife paper. Given the samples from 1044, we should probably not release collection year for discarded samples.

# Sample numbers

In [54]:
# Are there any samples with duplicates one of which was in v5 and the other of which is new?
v5_v6_groups = (
    tbl_panoptes_samples_final
    .selecteq('HasDuplicate', 'True')
    .valuecounts('IndividualGroup', 'InV5')
    .selecteq('count', 1)
    .sort(('IndividualGroup', 'InV5'))
    .distinct('IndividualGroup')
    .values('IndividualGroup')
    .array()
)
v5_v6_groups

array([ 650, 1353, 6614, 6683, 6687])

In [63]:
(
    tbl_panoptes_samples_final
    .selectin('IndividualGroup', v5_v6_groups)
    .cut(['Sample', 'OxfordSrcCode', 'Individual_ID', 'AlfrescoStudyCode', 'InV5', 'DiscardAsDuplicate', 'pc_pass_missing', 'IndividualGroup'])
    .displayall()
)

Sample 
 OxfordSrcCode 
 Individual_ID 
 AlfrescoStudyCode 
 InV5 
 DiscardAsDuplicate 
 pc_pass_missing 
 IndividualGroup 
 
 
 
 
 PH0027-C 
 CP268 
 CP268 
 1011-PF-KH-SU 
 True 
 True 
 0.34629377544337286 
 650 
 
 
 PH0027-Cx 
 CP268 
 CP268 
 1011-PF-KH-SU 
 False 
 False 
 0.0699845252924784 
 650 
 
 
 PT0229-C 
 H222 
 H222 
 1022-PF-MW-OCHOLLA 
 True 
 False 
 0.006057288523032272 
 1353 
 
 
 PT0229-Cx 
 H222 
 H222 
 1022-PF-MW-OCHOLLA 
 False 
 True 
 0.9979873248661213 
 1353 
 
 
 PJ0161-C 
 UNKNOWN_UFC_104 
 UFC_104 
 1155-PF-ID-PRICE 
 True 
 True 
 0.06205518242175962 
 6614 
 
 
 PJ0183-C 
 UFC_104 
 UFC_104 
 1155-PF-ID-PRICE 
 False 
 False 
 0.0387805159948118 
 6614 
 
 
 PJ0129-CW 
 UFM_26 
 UFM_26 
 1155-PF-ID-PRICE 
 False 
 True 
 0.042826092540035186 
 6683 
 
 
 PJ0129-Cx 
 UFM_26 
 UFM_26 
 1155-PF-ID-PRICE 
 True 
 False 
 0.03205287084719208 
 6683 
 
 
 PJ0134-C 
 UNKNOWN_UFM_32 
 UFM_32 
 1155-PF-ID-PRICE 
 True 
 False 
 0.013303432688232801 
 6687 
 
 
 PJ0253-C 
 UFM_32 
 UFM_32 
 1155-PF-ID-PRICE 
 False 
 True 
 0.025698930254658466 
 6687

In [56]:
tbl_panoptes_samples_final.pivot('InV5', 'DiscardAsDuplicate', 'DiscardAsDuplicate', len)

+---------+-------+------+
| InV5    | False | True |
+=========+=======+======+
| 'False' |  1030 |  159 |
+---------+-------+------+
| 'True'  |  5666 |  327 |
+---------+-------+------+

In [57]:
tbl_panoptes_samples_final.selectne('Site', 'Lab').pivot('InV5', 'DiscardAsDuplicate', 'DiscardAsDuplicate', len)

+---------+-------+------+
| InV5    | False | True |
+=========+=======+======+
| 'False' |   963 |  159 |
+---------+-------+------+
| 'True'  |  5658 |  326 |
+---------+-------+------+

In [58]:
tbl_panoptes_samples_final.selectne('Year', 'Lab').pivot('InV5', 'DiscardAsDuplicate', 'DiscardAsDuplicate', len)

+---------+-------+------+
| InV5    | False | True |
+=========+=======+======+
| 'False' |   963 |  159 |
+---------+-------+------+
| 'True'  |  5658 |  326 |
+---------+-------+------+

In [61]:
tbl_panoptes_samples_final.selectge('pc_genome_covered_at_1x', 0.5).selectne('Year', 'Lab').pivot('InV5', 'DiscardAsDuplicate', 'DiscardAsDuplicate', len)

+---------+-------+------+
| InV5    | False | True |
+=========+=======+======+
| 'False' |   893 |  150 |
+---------+-------+------+
| 'True'  |  5551 |  318 |
+---------+-------+------+

In [48]:
print(len(tbl_panoptes_samples_final.data()))
print(len(tbl_panoptes_samples_final.selectne('Year', 'Lab').data()))
print(len(tbl_panoptes_samples_final.selecteq('Year', 'Lab').data()))


7182
7106
76


# Variant numbers

In [27]:
callset = h5py.File(hdf_fn, mode='r')
variants = callset['variants']
PASS = callset['variants']['FILTER_PASS'][:]
is_snp = callset['variants']['is_snp'][:]

In [34]:
num_variants = collections.OrderedDict()
num_variants['Pass SNPs'] = np.count_nonzero(PASS & is_snp)
num_variants['Fail SNPs'] = np.count_nonzero(np.logical_not(PASS) & is_snp)
num_variants['Pass indels'] = np.count_nonzero(PASS & np.logical_not(is_snp))
num_variants['Fail indels'] = np.count_nonzero(np.logical_not(PASS) & np.logical_not(is_snp))
print(len(PASS))
print(len(is_snp))
num_variants

6051696
6051696


OrderedDict([('Pass SNPs', 1838733),
             ('Fail SNPs', 1329988),
             ('Pass indels', 1276027),
             ('Fail indels', 1606948)])

In [35]:
1838733+1329988+1276027+1606948

6051696

# Sample metadata

In [36]:
tbl_panoptes_samples_final.header()

('Sample',
 'OxfordSrcCode',
 'OxfordDonorCode',
 'Individual_ID',
 'AlfrescoStudyCode',
 'Type',
 'InV5.1',
 'InV5',
 'SubContinentCode',
 'SubContinentName',
 'RegionCode',
 'CountryCode',
 'CountryName',
 'Site',
 'Year',
 'Date',
 'HasDuplicate',
 'DiscardAsDuplicate',
 'run_accessions',
 'pc_pass_missing',
 'pc_genome_covered_at_1x',
 'Year_source',
 'sort_AlfrescoStudyCode',
 'IndividualGroup')

In [37]:
variables_to_count = [
 'AlfrescoStudyCode',
 'SubContinentCode',
 'SubContinentName',
 'RegionCode',
 'CountryCode',
 'CountryName',
 'Site',
 'Year']

In [44]:
for variable in variables_to_count:
    print(variable, len(tbl_panoptes_samples_final.selectne('Site', 'Lab').distinct(variable).data()))

AlfrescoStudyCode 52
SubContinentCode 9
SubContinentName 9
RegionCode 35
CountryCode 29
CountryName 29
Site 78
Year 16


In [47]:
tbl_panoptes_samples_final.selectne('Site', 'Lab').distinct('SubContinentName').cut('SubContinentName').displayall()

SubContinentName 
 
 
 
 
 Central Africa 
 
 
 East Africa 
 
 
 Eastern south-east Asia 
 
 
 Insular Africa 
 
 
 Oceania 
 
 
 South America 
 
 
 South Asia 
 
 
 West Africa 
 
 
 Western south-east Asia